libsolidity/interface/CompilerStack.cpp

```diff
@@ -425,9 +428,13 @@ bool CompilerStack::isRequestedContract(ContractDefinition const& _contract) con
 
 bool CompilerStack::compile()
 {
-	if (m_stackState < AnalysisSuccessful)
-		if (!parseAndAnalyze())
+	if (m_stackState < AnalysisPerformed)
+	{
+		if (!parse() && !m_parserErrorRecovery)
```

Comment:
Is there a way to leave out the `m_parserErrorRecovery`-check in the surrounding higher-level logic code?

Reply:
Of course there is a way to jiggle things around, but to do this cleanly really means rethinking the "compile" interface and possibly the State enumeration values.

Right now, `compile` returns a boolean indicating whether to continue or not. Something somewhere needs to indicate whether to continue if parsing fails.

If done in the callers, then that determination has to be done more times than just here.

The general pattern used is `if parse() && analyze()...`

This would by necessity become something like `parse(); testSomething() && analyze()..`

You seem to have strong ideas about what this should be, and it would probably be best if you just made the changes. But since I'd like to see this available to the public sooner, I can make the changes, if I could get the vision specified in a lot more detail so that I don't have to guess at things.

Reply:
Indeed, this is a drastic change to the compile interface and it requires drastic changes. The main requirement is that everything should be smooth and consistent afterwards.